project/MiMa.scala

```diff
@@ -735,7 +735,15 @@ object MiMa extends AutoPlugin {
 
         // #19828
         ProblemFilters.exclude[DirectAbstractMethodProblem]("akka.persistence.Eventsourced#ProcessingState.onWriteMessageComplete"),
-        ProblemFilters.exclude[ReversedAbstractMethodProblem]("akka.persistence.Eventsourced#ProcessingState.onWriteMessageComplete")
+        ProblemFilters.exclude[ReversedAbstractMethodProblem]("akka.persistence.Eventsourced#ProcessingState.onWriteMessageComplete"),
+
+        // #19834 rewrite Utf8Encoder with with GraphStages
+        ProblemFilters.exclude[FinalClassProblem]("akka.http.impl.engine.ws.Utf8Encoder"),
+        ProblemFilters.exclude[DirectMissingMethodProblem]("akka.http.impl.engine.ws.Utf8Encoder.inSurrogatePair"),
+        ProblemFilters.exclude[DirectMissingMethodProblem]("akka.http.impl.engine.ws.Utf8Encoder.onPush"),
+        ProblemFilters.exclude[DirectMissingMethodProblem]("akka.http.impl.engine.ws.Utf8Encoder.surrogateValue_="),
+        ProblemFilters.exclude[DirectMissingMethodProblem]("akka.http.impl.engine.ws.Utf8Encoder.onUpstreamFinish"),
+        ProblemFilters.exclude[DirectMissingMethodProblem]("akka.http.impl.engine.ws.Utf8Encoder.surrogateValue")
```

Comment:
Does filtering on package work?
That's great if so, lets apply that.
On Mar 24, 2016 14:15, "Alexander Golubev" <notifications@github.com> wrote:

> In project/MiMa.scala
> <https://github.com/akka/akka/pull/20085#discussion_r57314040>:
>
> > @@ -735,7 +735,15 @@ object MiMa extends AutoPlugin {
> >
> >          // #19828
> >          ProblemFilters.exclude[DirectAbstractMethodProblem]("akka.persistence.Eventsourced#ProcessingState.onWriteMessageComplete"),
> > -        ProblemFilters.exclude[ReversedAbstractMethodProblem]("akka.persistence.Eventsourced#ProcessingState.onWriteMessageComplete")
> > +        ProblemFilters.exclude[ReversedAbstractMethodProblem]("akka.persistence.Eventsourced#ProcessingState.onWriteMessageComplete"),
> > +
> > +        // #19834 rewrite Utf8Encoder with with GraphStages
> > +        ProblemFilters.exclude[FinalClassProblem]("akka.http.impl.engine.ws.Utf8Encoder"),
> > +        ProblemFilters.exclude[DirectMissingMethodProblem]("akka.http.impl.engine.ws.Utf8Encoder.inSurrogatePair"),
> > +        ProblemFilters.exclude[DirectMissingMethodProblem]("akka.http.impl.engine.ws.Utf8Encoder.onPush"),
> > +        ProblemFilters.exclude[DirectMissingMethodProblem]("akka.http.impl.engine.ws.Utf8Encoder.surrogateValue_="),
> > +        ProblemFilters.exclude[DirectMissingMethodProblem]("akka.http.impl.engine.ws.Utf8Encoder.onUpstreamFinish"),
> > +        ProblemFilters.exclude[DirectMissingMethodProblem]("akka.http.impl.engine.ws.Utf8Encoder.surrogateValue")
>
> yeah, need to filter out stream impl as well.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/akka/akka/pull/20085/files/0ef92be0ae4d1d1aa742b8c9b22a5f41fcb27a9c#r57314040>
>